File: xaml_masterdetail/CS/Data/Item.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MasterDetailApp.Data
{
    public class Item
    {
        public int Id { get; set; }
        public DateTime DateCreated { get; set; }
        public string Title { get; set; }
        public string Text { get; set; }
    }
}

File: xaml_masterdetail/CPP/Item.cpp
#include "pch.h"
#include "Item.h"

using namespace MasterDetailApp::Data;

Item::Item()
{
}

Item::Item(int id, Windows::Foundation::DateTime dateCreated, Platform::String^ title, Platform::String^ text)
	: m_id(id), m_dateCreated(dateCreated), m_title(title), m_text(text)
{
}